atproto-relay-docker/README.md

```diff
@@ -0,0 +1,147 @@
+- CPU: Intel Xeon-D 1520 - 4c/8t - 2.2 GHz/2.6 GHz
+- RAM: 32 GB ECC 2133 MHz
+- HDD: 2×2 TB HDD SATA
+
+Way more than what I need but only costs about $24 CAD a month.
+
+# Setup
+
+Once you have your machine ready to go the setup is relatively straightforward. Make sure to have ``shuf`` and ``parallel`` installed on your machine since it'll be needed for the last couple of steps.
+
+To start, clone the example setup from Tangled and change directory into the ``relay-docker`` project.
+
+```bash
+git clone git@tangled.sh:dane.is.extraordinarily.cool/relay-docker
+cd relay-docker
+```
+
+You'll then need to clone the ``indigo`` repository inside of the ``relay-docker``project.
+
+```bash
+git clone git@github.com:bluesky-social/indigo.git
+```
+
+## Environment Variables
+
+There is an ``.env.example`` file that you'll need to rename and fill in the information for.
+
+```bash
+mv .env.example .env
+```
+
+Then in your favourite text editor, open the ``.env`` file, we'll walk through what each variable does. For a full list of the available configurable environment variables, [head over to the indigo repository](https://github.com/bluesky-social/indigo/blob/main/cmd/relay/main.go#L48).
+
+``RELAY_LENIENT_SYNC_VALIDATION`` - If true, allow legacy upstreams which don't implement atproto sync v1.1. Would recommend to set this to ``true``
+
+``RELAY_REPLAY_WINDOW`` - The duration of output "backfill window", eg 24h. Basically how much data the Relay stores and for how long. ``48h`` is what I have mine set as and it takes up about 200-300GB of disk space.
+
+``RELAY_PERSIST_DIR`` - Where on the host machine the data is persisted. You can decide where but I've left it as ``/data/relay/persist``. 
+
+``RELAY_ADMIN_PASSWORD`` - this is the admin password you'll use to log in to the relay admin dashboard. Make sure to set something secure and save it somewhere safe.
+
+``RELAY_TRUSTED_DOMAINS`` - Which domains are "blessed", comma separated list. Typically means they have higher rate limits and won't get throttled by your relay. Should add ``*.host.bsky.network`` and other PDSes you know that you trust.
+
+``RELAY_HOST_CONCURRENCY`` - Number of concurrent worker routines per upstream host. This probably depends on your machine specs but I've configured mine at ``4``.
+
+Then you'll need to set up your environment variables for the database.
+
+``DATABASE_URL`` - postgres://relay:POSTGRES_PASSWORD_HERE@localhost:5432/relay
+
+``POSTGRES_USER`` - should be ``relay``
+
+``POSTGRES_PASSWORD`` - a secure password of your choosing, e.g ``openssl rand -base64 32``
+
+``POSTGRES_DB`` - also should be called ``relay``
+
+## Proxy
+
+This relay setup uses Caddy, but you can use whatever you like. Just make sure the settings match this.
+
+```bash
+yourdomain.com {
+    tls {
+      on_demand
+    }
+
+    handle /xrpc/com.atproto.sync.subscribeRepos {
+      reverse_proxy 127.0.0.1:2470 {
+        header_up Origin "yourdomain.com"
```

Comment:
Does the relay not have a trust proxy setting?

Reply:
Looked a little more into this, and no, the relay code doesn't appear to use https://echo.labstack.com/docs/ip-address